website/static/js/recentActivityWidget.js

```diff
@@ -0,0 +1,347 @@
+                    self.fileEvents = ((self.eventNumbers.files/self.totalEvents)*100 | 0) + (self.eventNumbers.files ? 5 : 0);
+                    self.commentEvents = ((self.eventNumbers.comments/self.totalEvents)*100 | 0) + (self.eventNumbers.comments ? 5 : 0);
+                    self.wikiEvents = ((self.eventNumbers.wiki/self.totalEvents)*100 | 0) + (self.eventNumbers.wiki ? 5 : 0);
+                    self.nodeEvents = ((self.eventNumbers.nodes/self.totalEvents)*100 | 0) + (self.eventNumbers.nodes ? 5 : 0);
+                    self.otherEvents = 100 - (self.fileEvents + self.commentEvents + self.wikiEvents + self.nodeEvents);
+                    self.cache = [];
+                }
+                if (!init) {
+                    self.cache.push(result.data.slice(0,9));
+                    self.cache.push(result.data.slice(10,19));
+                    if (!update) {
+                        self.activityLogs(self.cache[self.page - 1]);
+                    }
+                }
+                self.lastPage = (result.links.meta.total / (result.links.meta.per_page/2) | 0) + 1;
+            }, function _requestFail(){
+                self.errorLoading = true;
+                self.activityLogs([]);
+            });
+            return promise;
+        };
+
+        self.callLogs = function(filter) {
+            self.eventFilter = self.eventFilter === filter ? false : filter;
+            self.page = 1;
+            self.cache = [];
+            self.getLogs();
+        };
+
+        self.getLogs(true, false);
+
+        $(window).resize(function(){
+            var canvas = document.getElementById('rACanvas');
+            var ctx = canvas.getContext('2d');
+            ctx.clearRect(0, 0, canvas.width, canvas.height);
+        });
+
+        self.makeLine = function(canvas){
+            var ctx = canvas.getContext('2d');
+            ctx.clearRect(0, 0, canvas.width, canvas.height);
+            var progBar = self.select('#rAProgressBar');
+            var canvasElement = self.select('#rACanvas');
+            var handle = $('.ui-slider-handle');
+            var leftHandle = handle[0];
+            var rightHandle = handle[1];
+            ctx.beginPath();
+            ctx.moveTo(leftHandle.offsetLeft + (handle.width()/2), 0);
+            ctx.lineTo(progBar.offset().left - canvasElement.offset().left, self.canvasHeight);
+            ctx.strokeStyle = '#E0E0E0 ';
+            ctx.lineWidth = 2;
+            ctx.stroke();
+            ctx.beginPath();
+            ctx.moveTo(rightHandle.offsetLeft + (handle.width()/2), 0);
+            ctx.lineTo(progBar.offset().left + progBar[0].offsetWidth - canvasElement.offset().left, self.canvasHeight);
+            ctx.strokeStyle = '#E0E0E0 ';
+            ctx.lineWidth = 2;
+            ctx.stroke();
+        };
+        self.categoryColor = function(category){
+            if (category.indexOf('wiki') !== -1){ return '#d9534f'; }
```

Comment:
Unless I am incorrect, switch statements expect an exact string match, which this does not.